[MDS-4928] Assign initial status in incident creation in Core

areyeslo opened the pull request:
## Objective 

[MDS-4928](https://bcmines.atlassian.net/browse/MDS-4928)

https://bcmines.atlassian.net/browse/MDS-4970
https://bcmines.atlassian.net/browse/MDS-4928

- Replaced initial status code from `WNS`= Written Notice Submitted to any of the status codes `AFR` - Awaiting Final Report or `FRS` - FInal Report Submitted depending on the existence of Final Report.
-  `WNS` status code is not longer used. However, it is left in the database.
- The dangerous occurrence determined by the Mine is removed in Minespace-web and Core-web. Just the DO determined by the inspector is present.
- Uploading final report is optional in Minespace-web and core-web, can be done after submission (this would create an incident with initial status `AFR`.

areyeslo committed:
Assign initial status in incident creation in Core

areyeslo committed:
Added status to incident from Core-web

areyeslo committed:
Determination_type_code has precedence over mine_determination_type_code.

JustinMacaulay commented on services/core-web/src/components/mine/Incidents/MineIncident.js:
```diff
@@ -125,6 +126,36 @@ export class MineIncident extends Component {
+
+    if (!this.props.formValues.status_code) {
+      if (!determinationTypeCode) {
+        this.props.formValues.status_code = "WNS";
```
What are the possible statuses for an incident that is marked as a non dangerous occurrence? Are they always WNS and then Closed?

areyeslo replied:
That's correct., they are created as WNS and then go to the next stages which I guess are: Reviewing Severity Status/ Information Missing Status -> Under Review -> Closed.

areyeslo replied:
The logic completely changed. There is no WNS status code, but `AFR` or `FRS` code depending on the existence of the Final Report.

JustinMacaulay reviewed (commented):
just one question around status

areyeslo committed:
Remove relationship between Dangerous Occurrence determination and Final Report. Final Report has to be provided always provided.
Mine dangerous occurrence determination is removed. Just dangerous occurrence
determination made by inspector is kept

areyeslo committed:
Updating snapshot.

areyeslo committed:
Update snapshots.

simensma-fresh committed:
[MDS-4960] Fixed updating status of EORs through core (#2395)

MDS-4960 Fixed updating statsu of EORs through core

JustinMacaulay committed:
[MDS-4929] Minespace incident form fixes (#2389)

BenHarrington committed:
[MDS-4783] [CORE] Project Summary - Project lead and mrc_review_required fields to do not persist on save (#2392)

BenHarrington committed:
[MDS-4926] [MS] Major Project Submission Spatial File Upload failing for XML (#2393)

BenHarrington committed:
[MDS-4227] Inspector filter on the global incidents table doesn't work (#2396)

BenHarrington committed:
[FIX] Update transfer_mine_permit_xref_record function (#2273)

simensma-fresh committed:
[MDS-4891] Fixed expiring / expired notifications sending one day early (#2397)

* MDS-4891 Fixed expiring / expired notifications sending one day early

* MDS-4891 Fixed test

taraepp committed:
[MDS-4980] Codespaces Makefile changes (#2390)

quick commands to get workspace up and running, removed an outdated extension from recommendations and put docker in there instead

taraepp committed:
[MDS-4894] QP Expiring & Expired Notifications (#2398)

* Added notification for QP expiring, and for QP expiring in 60 days.
* eslint rule complaining about non-problems, disabled rule
* made the notification only show on minespace (EoR too), added enum so notifications can be sent to ms/core/all, default all
* Fixed tests: added factory so that we can look at MS subscriptions/access, then added a qp notification to the general tests, made adjustments to tests accordingly. Split up one test into two (testing MS/Core), but no tests specific to QP added yet
* Added some QP-specific tests, removed some junk I'd left in another file by accident

vyasworks committed:
[CHORE] build-fix - test with dummy email

vyasworks committed:
[CHORE] Upgrade checkout action to V3 (#2399)

taraepp committed:
[MDS-4883] consequence classification order (#2400)

* Change order of consequence in FE enums

* removed migration from afterMigrate script that was undoing my migration (it's possible that this has been adding back a value we've intended to remove) added the display_order to come from the BE for CC so we can potentially get all our info from one place. But also updated some FE enums that were used to determine the severity scale

* migrating NOD --> NRT for tsf records

* fixed data mocks for related tests

* update test snapshot to match data

matbusby-fw committed:
[MDS-4933] HSRC conditions (#2402)

updated HSRC conditions and how they are handled in Minespace and Core

taraepp committed:
[MDS-4994 ] Fix EoR/QP new appointment notification link (#2401)

* Had previously 'fixed' eor/qp link for expired and expiring, and it broke the new person notification. Fixing that so they should both work

matbusby-fw committed:
fixed incident_timestamp submission formatting

matbusby-fw committed:
Merge branch 'develop' into MDS-4928-ReportableIncidentsTestathonBugs

matbusby-fw committed:
updated IncidentForm.spec.js.snap
